slogr_test.go

```diff
@@ -0,0 +1,261 @@
+//go:build go1.21
+// +build go1.21
+
+/*
+Copyright 2023 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logr_test
```

Comment:
even if it is `func X() { return log.X() }`, the test seems redundant.

Reply:
I'll remove the test.

But it did find a problem earlier: instead of the `type SlogSink = logr.SlogSink`, I had a copy because I wanted it to be visible in the godoc. That normally works when methods are the same, but here the type is part of the interface, and that made the two interfaces different. Anyway, that's not a problem anymore.